stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -184,6 +184,50 @@ function eigen!(A::AbstractMatrix, C::Cholesky; sortby::Union{Function,Nothing}=
     GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
 end
 
+# Bunch-Kaufmann (LDLT) based solution for generalized eigenvalues and eigenvectors
+function eigen(A::AbstractMatrix{T}, B::BunchKaufman{T,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
```

Comment:
I suspect you need 
```suggestion
function eigen(A::AbstractMatrix{T}, B::BunchKaufman{T,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:BlasFloat}
```
here, don't you? And similarly below.

Reply:
Thank you for this comment. I agree. Upon noting that `BlasReal <: BlasFloat` and `BlasComplex <: BlasFloat`, I've changed `Number` to `BlasFloat` in all the four relevant method definitions. The change is in https://github.com/aravindh-krishnamoorthy/julia/commit/8b8f556d7095452e14a3bc7c46c4bc2f34d6bb9d.